cmd/kube-scheduler/app/server.go

```diff
@@ -339,3 +285,70 @@ func WithPlugin(name string, factory framework.PluginFactory) Option {
 		return registry.Register(name, factory)
 	}
 }
+
+// Setup creates a completed config and a scheduler based on the command args and options
+func Setup(ctx context.Context, args []string, opts *options.Options, outOfTreeRegistryOptions ...Option) (*schedulerserverconfig.CompletedConfig, *scheduler.Scheduler, error) {
+	if len(args) != 0 {
+		fmt.Fprint(os.Stderr, "arguments are not supported\n")
+	}
+
+	if errs := opts.Validate(); len(errs) > 0 {
+		return nil, nil, utilerrors.NewAggregate(errs)
+	}
+
+	if len(opts.WriteConfigTo) > 0 {
+		c := &schedulerserverconfig.Config{}
+		if err := opts.ApplyTo(c); err != nil {
+			return nil, nil, err
+		}
+		if err := options.WriteConfigFile(opts.WriteConfigTo, &c.ComponentConfig); err != nil {
+			return nil, nil, err
+		}
+		klog.Infof("Wrote configuration to: %s\n", opts.WriteConfigTo)
+		return nil, nil, nil
+	}
+
+	c, err := opts.Config()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Get the completed config
+	cc := c.Complete()
+
+	// Configz registration.
+	if cz, err := configz.New("componentconfig"); err == nil {
```

Comment:
Maybe part of `Run` as well?

Reply:
yes, we can, but I think this is more like a config "setup"

Reply:
Well, this registers the component config in the /configz server. It comes down to the question in https://github.com/kubernetes/kubernetes/pull/88728#discussion_r387746862

Reply:
moved to `Run`